draft/mson-parameters-headers.md

```diff
@@ -0,0 +1,216 @@
+1. [Description of enumeration elements](https://github.com/apiaryio/api-blueprint/issues/119)
+1. [Array in parameters](https://github.com/apiaryio/api-blueprint/issues/240)
+1. [Minimum/Maximum values for URI Parameters](https://github.com/apiaryio/api-blueprint/issues/211)
+
+    NOTE: This is going to be addressed once MSON introduces validation.
+
+### HTTP Headers Issues Tracked
+
+This RFC addresses following issues:
+
+1. [Parametric Headers Description](https://github.com/apiaryio/api-blueprint/issues/26)
+1. [Referenced HTTP headers](https://github.com/apiaryio/api-blueprint/issues/17)
+1. [Support for multiline syntax header](https://github.com/apiaryio/api-blueprint/issues/60)
+
+In addition this issue is solved by this RFC:
+
+1. [Reuse headers and attributes](https://github.com/apiaryio/api-blueprint/issues/259)
+
+## Implementation
+
+### Parameters
+
+The implementation should assume the existing syntax for API Blueprint
+[URI parameters section][] is, in fact, an [MSON type declaration][].
+
+The type being declared inherits from a *Parameters base type* unless specified
+otherwise. If another base type is specified it must inherit from the
+*Parameters base type*.
+
+The *Parameter base type* may the [API Description Namespace][]
+[`Href Variables`][] type.
+
+The *Parameters base type* should be available as a type for use within any
+[data structures section][].
+
+#### Example
+
+```apib
+# Resource [/resource/{p}{?q}]
++ Parameters
+    + p
+    + q
+```
+
+Assuming `Parameters` as the name of the *Parameters base type*:
+
+```apib
+# Resource [/resource/{p}{?q}]
++ Parameters (My Parameters)
+
+# Data Structures
+## My Parameters (Parameters)
++ p
++ q
+```
+
+### Headers
+
+The implementation of this RFC should introduce the use of MSON syntax in the
+[Headers section][]. Similarly to Parameters section this Headers section should
+be considered as [MSON type declaration][].
+
+The type being declared inherits from a *Headers base type* unless specified
+otherwise. If another base type is specified it must inherit from the
+*Headers base type*.
+
+The *Headers base type* may the [API Description Namespace][] [`HTTP Headers`][]
+type.
+
+The *Headers base type* should be available as a type for use within any
+[data structures section][].
+
+### Example
+
+```apib
+# Resource [/resource]
+## Retrieve [GET]
++ Response 200
+    + Headers
+        + Server: nginx
+        + `X-RateLimit-Remaining`: 4987 (number) - Remaining rate limit
+```
+
+Assuming `Headers` as the name of the *Headers base type*:
+
+```apib
+# Resource [/resource]
+## Retrieve [GET]
++ Response 200
+    + Headers (My Headers)
+
+# Data Structures
+## My Headers
++ Server: nginx
++ `X-RateLimit-Remaining`: 4987 (number) - The rate limit remaining
+```
+
+## Considerations
+
+### Parameters base type constraints
+
+Using any type in as a member type in a descendant of Parameters base
+type may lead into problems while representing its value in URL. The
```

Comment:
Yeah, the objects and nested arrays serialisation is going to be complex and there is no set rules anywhere on how people do this. Each programming language & framework does it in it's own style